src/libAtomVM/nifs.c

```diff
@@ -3651,6 +3658,54 @@ static term nif_erlang_make_fun_3(Context *ctx, int argc, term argv[])
     return term_make_function_reference(module_term, function_term, arity_term, &ctx->heap);
 }
 
+static term nif_erlang_fun_info_2(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+    term fun = argv[0];
+    term key = argv[1];
+
+    VALIDATE_VALUE(fun, term_is_fun);
+    VALIDATE_VALUE(key, term_is_atom);
+
+    term value;
+    switch (key) {
+        case MODULE_ATOM: {
+            term module_name;
+            term_get_function_mfa(fun, &module_name, NULL, NULL, ctx->global);
+            value = module_name;
```

Comment:
Are we fine with those value terms being uninitialized before `term_get_function_mfa` call? I tend to do that when you assign it immediately but I can why it may not be desired when it's done in the function call somewhere.

Reply:
There is no point about initializing output variables, so it is ok like this.